modules/core/src/main/java/org/apache/ignite/internal/util/nio/GridTcpNioCommunicationClient.java

```diff
@@ -157,22 +161,24 @@ public GridNioSession session() {
         return S.toString(GridTcpNioCommunicationClient.class, this, super.toString());
     }
 
-    public void sendHeartbeatsIfNeeded() {
-        if (!useHeartbeats)
+    /**
+     * Sends special {@link ConnectionCheckMessage} through the channel to check if connection still alive.
+     */
+    public void checkConnection() {
```

Comment:
Maybe `checkConnectionIfEnabled()` (or something similar, to help a reader understand that it's actually a conditional send)?

Reply:
ok